IPython/core/interactiveshell.py

```diff
@@ -902,42 +903,45 @@ def init_virtualenv(self):
         if 'VIRTUAL_ENV' not in os.environ:
             # Not in a virtualenv
             return
-        
-        p = os.path.normcase(sys.executable)
-        p_venv = os.path.normcase(os.environ['VIRTUAL_ENV'])
-
-        # executable path should end like /bin/python or \\scripts\\python.exe
-        p_exe_up2 = os.path.dirname(os.path.dirname(p))
-        if p_exe_up2 and os.path.exists(p_venv) and os.path.samefile(p_exe_up2, p_venv):
-            # Our exe is inside the virtualenv, don't need to do anything.
+        elif os.environ['VIRTUAL_ENV'] == '':
```

Comment:
I'm changing this because of this issue https://github.com/ipython/ipython/issues/11100
which was resolved this way https://github.com/ipython/ipython/pull/11172

The code I'm using here was suggested on the issue, and it seems better to me,
specially now that we are using Pathlib, since Path('') actualy means the path in which you currently are

So, as is, we would get a True when we should get a False
In [4]: Path('').exists()
Out[4]: True

If the warning is a bit excessive I can remove it and merge with the upwards if statement